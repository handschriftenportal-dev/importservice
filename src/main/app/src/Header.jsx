import React from "react";
import {Grid, Hidden, Typography} from "@material-ui/core";
import {ImportHeader} from "./css/ImportHeader";

function Header(props) {
  return <ImportHeader component="header">
    <Grid container justify="center" alignItems="center"
          direction="column">
      <Hidden>
        <Grid item md={1}></Grid>
      </Hidden>
      <Grid item xs={12} md={10}>
        <Typography id="uploadheadline" variant="h3"
                    component="h1">{props.title}</Typography>
      </Grid>
      <Hidden>
        <Grid item md={1}></Grid>
      </Hidden>
    </Grid>
  </ImportHeader>
}

export default Header;
