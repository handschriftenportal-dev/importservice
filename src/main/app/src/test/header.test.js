import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import Header from "../Header";

describe('Header Import Testsuite', () => {

  let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("Test Header Rendering", () => {
    act(() => {
      render(<Header title="Handschriftenportal Erfassung"/>, container);
    });
    expect(container.textContent).toBe("Handschriftenportal Erfassung");
  })

  it('Header Snapshot Test', () => {
    const tree = renderer
    .create(<Header title="Handschriftenportal Erfassung"/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
