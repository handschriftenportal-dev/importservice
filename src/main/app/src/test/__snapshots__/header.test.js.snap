// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Import Testsuite Header Snapshot Test 1`] = `
<header
  className="Styled(MuiBox)-root-135"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
    >
      <h1
        className="MuiTypography-root MuiTypography-h3"
        id="uploadheadline"
      >
        Handschriftenportal Erfassung
      </h1>
    </div>
  </div>
</header>
`;
